common/test/acceptance/tests/lms/test_conditional.py

```diff
@@ -0,0 +1,134 @@
+        metadata = {}
+        source_block = None
+        if block_type == 'problem':
+            problem_factory = StringResponseXMLFactory()
+            problem_xml = problem_factory.build_xml(
+                question_text='The answer is "correct string"',
+                case_sensitive=False,
+                answer='correct string',
+            ),
+            problem = XBlockFixtureDesc('problem', 'Test Problem', data=problem_xml[0])
+            metadata = {
+                'xml_attributes': {
+                    'attempted': 'True'
+                }
+            }
+            source_block = problem
+        elif block_type == 'poll':
+            poll = XBlockFixtureDesc(
+                'poll_question',
+                'Conditional Poll',
+                question='Is this a good poll?',
+                answers=[
+                    {'id': 'yes', 'text': POLL_ANSWER},
+                    {'id': 'no', 'text': 'Of course not!'}
+                ],
+            )
+            metadata = {
+                'xml_attributes': {
+                    'poll_answer': 'yes'
+                }
+            }
+            source_block = poll
+        else:
+            raise NotImplementedError()
+
+        course_fixture.create_xblock(vertical.locator, source_block)
+        # create conditional
+        conditional = XBlockFixtureDesc(
+            'conditional',
+            'Test Conditional',
+            metadata=metadata,
+            sources_list=[source_block.locator],
+        )
+        result_block = XBlockFixtureDesc(
+            'html', 'Conditional Contents',
+            data='<html><div class="hidden-contents">Hidden Contents</p></html>'
+        )
+        course_fixture.create_xblock(vertical.locator, conditional)
+        course_fixture.create_xblock(conditional.locator, result_block)
+
+    def test_conditional_hides_content(self):
+        self.install_course_fixture()
+        self.courseware_page.visit()
+        conditional_page = ConditionalPage(self.browser)
+        self.assertFalse(conditional_page.is_content_visible())
+
+    def test_conditional_displays_content(self):
+        self.install_course_fixture()
+        self.courseware_page.visit()
+        # Answer the problem
+        problem_page = ProblemPage(self.browser)
+        problem_page.fill_answer('correct string')
+        problem_page.click_check()
+        self.courseware_page.visit()
+        # Verify that we can see the content.
+        conditional_page = ConditionalPage(self.browser)
+        self.assertTrue(conditional_page.is_content_visible())
+
+    def test_conditional_handles_polls(self):
+        self.install_course_fixture(block_type='poll')
+        self.courseware_page.visit()
+        # Fill in the conditional page poll
+        conditional_page = ConditionalPage(self.browser)
+        conditional_page.fill_in_poll()
+        # Refresh page and ensure the page is visible.
+        self.courseware_page.visit()
```

Comment:
Same here..I'm wondering if we could avoid a reload. In part it's test performance, in part it's representing what the user would do...